AutomatedTesting/Gem/PythonTests/scripting/ScriptCanvasComponent_OnEntityActivatedDeactivated_PrintMessage.py

```diff
@@ -58,101 +63,74 @@ class ScriptCanvasComponent_OnEntityActivatedDeactivated_PrintMessage():
-
-        """
-
-        entity = scripting_tools.validate_entity_exists_by_name(entity_tuple["name"], test)
-        state_correct = scripting_tools.validate_entity_start_state_by_name(entity_tuple["name"], entity_tuple["status"])
-
-        return state_correct
-
-    def validate_test_entities(self):
-
-        entities_valid = (self.validate_entity_state(activated_dict, Tests.activated_exists) and
-                        self.validate_entity_state(deactivated_dict, Tests.deactivated_exists) and
-                        self.validate_entity_state(controller_dict, Tests.controller_exists))
-
-        Report.critical_result(Tests.start_states_correct, entities_valid)
-
-
-    @pyside_utils.wrap_async
-    async def run_test(self):
-
-        # 1) Create temp level
-        general.idle_enable(True)
-        hydra.open_base_level()
-        helper.wait_for_condition(lambda: general.get_current_level_name() == BASE_LEVEL_NAME, WAIT_TIME_3)
-        general.close_pane("Error Report")
-
-        # 2) create all the entities we need for the test
-        self.setup_level_entities()
-
-        # 3) Validate the entities were created and configured
-        self.validate_test_entities()
-
-        # 4) Start the Tracer
-        with Tracer() as section_tracer:
-
-            # 5) Enter Game Mode
-            helper.enter_game_mode(Tests.game_mode_entered)
-
-            # 6 Wait one second for graph timers then exit game mode
-            general.idle_wait(WAIT_ONE_SECOND)
-            helper.exit_game_mode(Tests.game_mode_exited)
-
-            # 7) Validate Print message
-            found_expected_lines = scripting_tools.located_expected_tracer_lines(self, section_tracer, EXPECTED_LINES)
-            helper.wait_for_condition(lambda: found_expected_lines is not None, WAIT_TIME_3)
-
-        Report.result(Tests.lines_found, found_expected_lines)
-
-
-
-test = ScriptCanvasComponent_OnEntityActivatedDeactivated_PrintMessage()
-test.run_test()
+    import os
+    import azlmbr.paths as paths
+    import scripting_utils.scripting_tools as scripting_tools
+    from editor_python_test_tools.utils import TestHelper as TestHelper
+    from editor_python_test_tools.utils import Report, Tracer
+    import azlmbr.legacy.general as general
+    from scripting_utils.scripting_constants import (WAIT_TIME_3, WAIT_TIME_1)
+
+    controller_dict = {
+        "name": "Controller",
+        "status": "active",
+        "path": os.path.join(paths.projectroot, "ScriptCanvas", "OnEntityActivatedScripts", "controller.scriptcanvas"),
+    }
+    activated_dict = {
+        "name": "ActivationTest",
+        "status": "inactive",
+        "path": os.path.join(paths.projectroot, "ScriptCanvas", "OnEntityActivatedScripts", "activator.scriptcanvas"),
+    }
+    deactivated_dict = {
+        "name": "DeactivationTest",
+        "status": "active",
+        "path": os.path.join(paths.projectroot, "ScriptCanvas", "OnEntityActivatedScripts", "deactivator.scriptcanvas"),
+    }
```

Comment:
These might be a good candidate to be tuples instead of dictionaries since they are only 1 entry deep. This will allow for tuple unpacking in the helper function for cleaner code.

Also, if we remove these from the test function to elsewhere in the file, it'll help with readability of test portion of the file.

Reply:
Additionally, a more meaningful name would be *_entity instead of dict.

Reply:
I personally don't like the use of dictionaries here but I'm having trouble seeing how tuples would be better. If I remember tuples are a collection type that have a format like a list or array with the benefit of not having to contain objects of the same type; `{"hello, 40, True, } `. It's not clear to me why we would use a tuple here or how to effectively use it.

Reply:
When you unpack the dictionaries in the `setup_level_entities` method, you can assign them to meaningful names.

EG: `name, status, path = entity` then use those directly instead of having mystery strings accessing the dictionary. 

Another alternative would be to use a [dataclass](https://docs.python.org/3/library/dataclasses.html).

Lastly, when building these, since you're not using these values anywhere else, you can listify them on generation.

Below is an example of combining a things together using the dataclass variant.

```suggestion
SCRIPT_CANVAS_DEFAULT_LOCATION= os.path.join(paths.projectroot, "ScriptCanvas")

def test_file_path(file_name: str) -> os.Pathlike:
    return os.path.join(SCRIPT_CANVAS_DEFAULT_LOCATION, "OnEntityActivatedScripts", file_name)
    
@dataclass
class testEntity:
    def __init__(self, name: str, status: str, path: os.Pathlike):
        self.name = name
        self.status = status
        self.path = path

test_entities = [
    testEntity("Controller", "active", test_file_path("controller.scriptcanvas"),
    testEntity("ActivationTest", "inactive", test_file_path("activator.scriptcanvas"),
    testEntity("DeactivationTest", "active", test_file_path("deactivator.scriptcanvas")
]

setup_level_entities(test_entities)
```

Reply:
took this example and implemented it. There's a small issue with the way test runner handles imports so the os and paths imports need to be within the class constructor but otherwise this works.